source/Nevermore.IntegrationTests/CustomTypes/CustomSingleTypeInColumnFixture.cs

```diff
@@ -0,0 +1,114 @@
+                var read = transaction.Query<CustomTypeWithColumnToTestSerialization>()
+                    .FirstOrDefault();
+                read.CustomTypeUsedAsAProperty.Should().Be(JsonConvert.SerializeObject(model.CustomTypeUsedAsAProperty));
+            }
+        }
+
+        [Test]
+        public void ShouldBeAbleToReadBackFromColumn()
+        {
+            using (var transaction = Store.BeginTransaction())
+            {
+                var model = new ModelTypeWithColumn
+                {
+                    CustomTypeUsedAsAProperty =  new CustomTypeUsedAsAProperty
+                    {
+                        Name = "bar",
+                        Value = 15
+                    }
+                };
+                transaction.Insert(model);
+
+                var read = transaction.Query<ModelTypeWithColumn>()
+                    .FirstOrDefault();
+                read.Should().NotBeSameAs(model);
+                read.CustomTypeUsedAsAProperty.Should().NotBeSameAs(model.CustomTypeUsedAsAProperty);
+                read.CustomTypeUsedAsAProperty.Name.Should().Be("bar");
+                read.CustomTypeUsedAsAProperty.Value.Should().Be(15);
+            }
+        }
+
+        protected override IEnumerable<DocumentMap> AddCustomMappingsForSchemaGeneration()
+        {
+            return new DocumentMap[]
+            {
+                new ModelTypeWithColumnMap()
+            };
+        }
+
+        protected override IEnumerable<DocumentMap> AddCustomMappings()
+        {
+            return new DocumentMap[]
+            {
+                new ModelTypeWithColumnMap(),
+                new CustomTypeWithColumnToTestSerializationMap()
+            };
+        }
+
+        protected override IEnumerable<CustomTypeDefinition> CustomTypeDefinitions()
+        {
+            return new[]
+            {
+                new CustomTypeUsedAsAPropertyCustomTypeDefinition()
+            };
+        }
+
+        class ModelTypeWithColumn : IId
+        {
+            public string Id { get; set; }
+            public CustomTypeUsedAsAProperty CustomTypeUsedAsAProperty { get; set; }
+        }
+
+        class ModelTypeWithColumnMap : DocumentMap<ModelTypeWithColumn>
+        {
+            public ModelTypeWithColumnMap()
+            {
+                Column(x => x.CustomTypeUsedAsAProperty);
+            }
+        }
+
+        class CustomTypeWithColumnToTestSerialization : IId
+        {
+            public string Id { get; set; }
+            public string CustomTypeUsedAsAProperty { get; set; }
+            public string JSON { get; set; }
+        }
+
+        class CustomTypeWithColumnToTestSerializationMap : DocumentMap<CustomTypeWithColumnToTestSerialization>
+        {
+            public CustomTypeWithColumnToTestSerializationMap()
+            {
+                TableName = "ModelTypeWithColumn";
+                
+                Column(m => m.CustomTypeUsedAsAProperty);
+                Column(m => m.JSON);
```

Comment:
Mapping the `JSON` column shouldn't be necessary?

Reply:
This is a pattern that a number of the tests in here use, they map the JSON as a string and then compare to ensure it's exactly what's expected. This actually showed up some quirks in things I was testing, where the raw JSON showed some nested objects were being serialized as strings rather than objects, which all worked with new data going in/out but would fail with legacy data.